### PR TITLE
API to get tenants credentials from env variables

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/NEX-780

depends on https://github.com/oat-sa/generis/pull/753/files

see *readme.md* for samples

### CODE REVIEW

File: model/api/TenantDriverApiReader.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
+<?php
+/**
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/api/TenantDriverApiReader.php]: may i ask you to run the 
```
phpcbf . --extensions=php --standard=PSR12
```

File: README.md [hunk 2]
```diff
@@ -8,3 +8,66 @@ RDS queue for prod-like is the must, needs to be configured as per documentation
 To be independent workers has be to launched as a separate processes: 
 - `sudo -u www-data php index.php 'oat\taoTaskQueue\scripts\tools\RunWorker' --queue=ng_publication` 
 - `sudo -u www-data php index.php 'oat\taoTaskQueue\scripts\tools\RunWorker' --queue=queue`
+
+
+#### TanantAPI configuration
+
+There are 3 implementations exists to work with a Tenant API,  configurable at `config/taoDeliverConnect/DeliverTenantService.conf.php`
```
Comment (mike-ionut-mihai-sandu-tao) [inline README.md]:
```suggestion
There are 3 existing implementations that work with the Tenant API configurable at `config/taoDeliverConnect/DeliverTenantService.conf.php`
```

File: README.md [hunk 3]
```diff
@@ -8,3 +8,66 @@ RDS queue for prod-like is the must, needs to be configured as per documentation
...
+
+1. `\oat\taoDeliverConnect\model\api\TenantHttpAPI` used to retrieve data based on agreed Rest contract
```
Comment (mike-ionut-mihai-sandu-tao) [inline README.md]:
```suggestion
1. `\oat\taoDeliverConnect\model\api\TenantHttpAPI` is used to retrieve data based on agreed Rest contract
```

File: README.md [hunk 4]
```diff
@@ -8,3 +8,66 @@ RDS queue for prod-like is the must, needs to be configured as per documentation
...
+   
+   
+        DELIVERTENANT_COUNT=2
+         
+        DELIVERTENANT_0_ID=1
+        DELIVERTENANT_0_LABEL=coool_label
+        DELIVERTENANT_0_CUSTOMER_ID=77
+        DELIVERTENANT_0_ROOT_URL='http://deliver.docker.localhost/api/v1/'
+         
+        DELIVERTENANT_0_LTI_CREDENTIALS_COUNT=2
```
Comment (mike-ionut-mihai-sandu-tao) [inline README.md]: maybe we can add some comments that will ease up the work for the next developer

File: manifest.php [hunk 5]
```diff
@@ -29,10 +29,10 @@
     'label' => 'Deliver Connect',
     'description' => 'Extension to manage the connection between Tao Backoffice and Nextgen Deliver',
     'license' => 'GPL-2.0',
-    'version' => '3.4.0',
+    'version' => '3.5.0',
     'author' => 'Open Assessment Technologies SA',
     'requires' => array(
-        'generis' => '>=12.15.0',
+        'generis' => '>=12.16.0',
```
Comment (bartlomiejmarszal) [inline manifest.php]: Is this necessary?

Comment (boajer) [reply manifest.php]: The KV driver to access ENV variables, needed for the functional purpose of this PR is introduced in generis 12.16
There are no direct code dependecy in this PR so far, but if we add some script to setup or configure service in the expected way - this dependcy is the must.

File: model/api/TenantDriverApiReader.php [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+     * @throws common_exception_InconsistentData
+     */
+    private function readLtiCredentials($name, array $parentLevel = [0])
+    {
+        $result = [];
+
+        $length = $this->readSingleValue($name, $parentLevel);
+        if (isset($length) && is_numeric($length)) {
+            for ($i = 0; $i < $length; $i++) {
+                $depth = array_merge($parentLevel, [$i]);
```
[the remark this answers is not in the document]
Comment (boajer) [reply model/api/TenantDriverApiReader.php]: What would you recommend instead?

[the remark this answers is not in the document]
Comment (mike-ionut-mihai-sandu-tao) [reply model/api/TenantDriverApiReader.php]: for me it look more like a sum algorithm that your doing at line 89 and line 98
and i think a sum will be faster than array merge

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**